sdk/tests/conformance2/rendering/vertex-id.html

```diff
@@ -0,0 +1,239 @@
+    test(1, 2);
+    test(10000, 2);
+    test(100000, 2);
+    test(1000000, 2);
+
+    const INT32_MAX = 0x7fffffff;
+
+    test = function(first, count) {
+        debug("");
+        debug(`drawArrays(first: ${first}, count: ${count})`);
+        gl.drawArrays(gl.POINTS, first, count);
+        if (!wtu.glErrorShouldBe(gl, [gl.NO_ERROR, gl.OUT_OF_MEMORY])) {
+            ensureVal("", first+count-1);
+        }
+    };
+
+    test(INT32_MAX-2, 1);
+    test(INT32_MAX-1, 1);
+    test(INT32_MAX, 1);
+
+    // -
+
+    debug("");
+    debug("----------------");
+    debug("drawElements");
+
+    const indexBuffer = gl.createBuffer();
+    gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, indexBuffer);
+    const indexData = new Uint16Array([1, 2, 5, 3, 10000]);
+    debug("indexData: " + indexData);
+    gl.bufferData(gl.ELEMENT_ARRAY_BUFFER, indexData, gl.STATIC_DRAW);
+
+    test = function(first, count) {
+        debug("");
+        debug(`drawElements(first: ${first}, count: ${count})`);
+        gl.drawElements(gl.POINTS, count, gl.UNSIGNED_SHORT, first*2);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR);
+        ensureVal("", indexData[first+count-1]);
+    };
+
+    for (let f = 0; f < indexData.length; ++f) {
+        for (let c = 1; f + c <= indexData.length; ++c) {
+            test(f, c);
+        }
+    }
+
+    // -
+
+    debug("");
+    debug("----------------");
+    debug("Via transform feedback");
+
+    gl.transformFeedbackVaryings(prog, ["vVertexID"], gl.INTERLEAVED_ATTRIBS);
+    wtu.linkProgram(gl, prog);
+    gl.useProgram(prog);
+
+    const tfBuffer = gl.createBuffer();
+    gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 0, tfBuffer);
+    gl.bufferData(gl.TRANSFORM_FEEDBACK_BUFFER, 4*10000, gl.DYNAMIC_READ);
+
+    test = function(offset, count) {
+        debug("");
+        debug("drawArrays(" + offset + ", " + count + ")");
+        gl.beginTransformFeedback(gl.POINTS);
+        gl.drawArrays(gl.POINTS, offset, count);
+        gl.endTransformFeedback();
+        gl.getBufferSubData(gl.TRANSFORM_FEEDBACK_BUFFER, 0, readValData);
+        let ok = true;
+        for (let i = 0; i < readValData.length; i++) {
+            if (i >= count)
+                break;
+            const expected = offset + i;
+            const was = readValData[i];
+            if (was != expected) {
+                testFailed("[" + i + "] expected " + expected + ", was " + was);
+                ok = false;
+                break;
+            }
+        }
+        if (ok) {
+            testPassed("ok");
+        }
+    };
+
+    test(0, 1);
+    test(1, 1);
+    test(10000, 1);
+
+    test(0, 2);
+    test(1, 2);
+    test(10000, 2);
+
+    test(10000, 10000);
+
+    // -
+
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "There should be no remaining errors");
+})();
+
+debug("");
+var successfullyParsed = true;
+</script>
+<script src="../../js/js-test-post.js"></script>
+
+</body>
+</html>
+
+<!--
+
+/*
+** Copyright (c) 2013 The Khronos Group Inc.
```

Comment:
Please move closer to the top of the file and update the copyright date.

Reply:
Let's talk briefly about this on the concall tomorrow.